dotnet/src/Extensions/PromptTemplate.Handlebars/Helpers/KernelHelpers/KernelFunctionHelpers.cs

```diff
@@ -0,0 +1,210 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using HandlebarsDotNet;
+using HandlebarsDotNet.Compiler;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+
+namespace Microsoft.SemanticKernel.PromptTemplate.Handlebars.Helpers;
+
+/// <summary>
+/// Utility class for registering kernel functions as helpers in Handlebars.
+/// </summary>
+internal static class KernelFunctionHelpers
+{
+    /// <summary>
+    /// Register all (default) or specific categories.
+    /// </summary>
+    /// <param name="handlebarsInstance">The <see cref="IHandlebars"/>-context.</param>
+    /// <param name="kernel">Kernel instance.</param>
+    /// <param name="executionContext">Kernel arguments maintained as the executing context.</param>
+    /// <param name="nameDelimiter">The character used to delimit the plugin name and function name in a Handlebars template.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    public static void Register(
+        IHandlebars handlebarsInstance,
+        Kernel kernel,
+        KernelArguments executionContext,
+        string nameDelimiter,
+        CancellationToken cancellationToken = default)
+    {
+        foreach (var function in kernel.Plugins.GetFunctionsMetadata())
+        {
+            RegisterFunctionAsHelper(kernel, executionContext, handlebarsInstance, function, nameDelimiter, cancellationToken);
+        }
+    }
+
+    #region private
+
+    private static void RegisterFunctionAsHelper(
+        Kernel kernel,
+        KernelArguments executionContext,
+        IHandlebars handlebarsInstance,
+        KernelFunctionMetadata functionMetadata,
+        string nameDelimiter,
+        CancellationToken cancellationToken = default)
```

Comment:
As a good practice to follow, use default cancellation tokens only for public APIs and not for internal ones. This ensures proper passing of the token from the public API down to lower levels, preventing bugs. The API consumer code may decide whether to pass it or not, while internally, it will be passed through all levels.

Reply:
heard, removed all ` = default` on cancellation tokens here